schoolopy/authentication.py

```diff
@@ -0,0 +1,135 @@
+import random
+import time
+import yaml
+import requests_oauthlib
+import json
+from urllib import parse
+import requests
+
+class SchoologyAuth:
+    def __init__(self, consumer_key, consumer_secret, user_id=None):
+        self.API_ROOT = 'https://api.schoology.com/v1'
+        self.DOMAIN_ROOT = 'https://fccps.schoology.com'
+
+        self.user_id = user_id
+        self.consumer_key = consumer_key
+        self.consumer_secret = consumer_secret
+
+        self.oauth = requests_oauthlib.OAuth1Session(self.consumer_key, self.consumer_secret)
+
+        if user_id != None:
+            with open('auth.yml', 'r+') as f:
```

Comment:
This sort of thing shouldn't be hardcoded; maybe figure out how to get this data through function parameters?